src/w3iProxy/w3iChatFacade.ts

```diff
@@ -35,6 +47,58 @@ class W3iChatFacade implements W3iChat {
 
     const ProviderClass = this.providerMap[this.providerName]
     this.provider = new ProviderClass(this.emitter)
+
+    // Discuss expiry of messages
+    this.messageReplaySubject = new ReplaySubject(undefined, ONE_DAY / 2)
+    this.handleMessagePublishing()
+  }
+
+  private handleMessagePublishing() {
+    // Stop trying to resend the message after 3 tries.
+    this.messageReplaySubject.pipe(filter(params => params.count < 3)).subscribe(params => {
+      from(this.provider.message(params))
+        .pipe(
+          timeout({
+            first: this.messageSendTimeout,
+            with: () => throwError(() => new Error())
+          })
+        )
+        .subscribe({
+          next: () => {
+            this.emitter.emit('chat_message_sent', {
+              ...params
+            })
+          },
+          error: () => {
+            /*
+             * Create arbitrary delay.
+             * The timestamp is updated to be the current time.
+             */
+            setTimeout(() => {
+              this.messageReplaySubject.next({
+                ...params,
+                count: params.count + 1,
+                timestamp: Date.now()
+              })
+            }, this.messageSendTimeout * 1.25)
```

Comment:
Why do we need a delay here and why the magic number `1.25` as a multiplier for what seems to be an already defined timeout value?

Reply:
Just wanted it to be longer than the time of the `timeout` function, `1.25` is entirely arbitrary